apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+    */
+    function _onDisputableActionAllowed(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Active;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit ResumeVote(_voteId);
+    }
+
+    /**
+    * @dev Reject a vote
+    * @param _voteId Identification number of the vote to be rejected
+    */
+    function _onDisputableActionRejected(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Cancelled;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit CancelVote(_voteId);
+    }
+
+    /**
+    * @dev Void an entry
+    * @param _voteId Identification number of the entry to be voided
+    */
+    function _onDisputableActionVoided(uint256 _voteId) internal {
+        _onDisputableActionAllowed(_voteId);
+    }
+
+    /**
+    * @dev Internal function to check if a vote can be executed. It assumes the queried vote exists.
+    * @return True if the given vote can be executed, false otherwise
+    */
+    function _canExecute(Vote storage vote_) internal view returns (bool) {
+        // If vote is already executed, it cannot be executed again
+        if (vote_.executed) {
+            return false;
+        }
+
+        // If the vote is paused or cancelled, it cannot be executed
+        if (_isPausedOrCancelled(vote_)) {
+            return false;
+        }
+
+        // If the vote is still open, it cannot be executed
+        if (_isVoteOpen(vote_)) {
+            return false;
+        }
+
+        // If the vote does not have enough support, it cannot be executed
+        uint256 totalVotes = vote_.yea.add(vote_.nay);
+        if (!_isValuePct(vote_.yea, totalVotes, vote_.supportRequiredPct)) {
+            return false;
+        }
+
+        // If the vote has not reached min quorum, it cannot be executed
+        if (!_isValuePct(vote_.yea, vote_.votingPower, vote_.minAcceptQuorumPct)) {
+            return false;
+        }
+
+        // If the vote cannot proceed due to an Agreement dispute, it cannot be executed
+        if (!_canProceedAgreementAction(vote_.actionId)) {
+            return false;
+        }
+
+        // If non of the above conditions are met, it can be executed
+        return true;
+    }
+
+    /**
+    * @dev Internal function to check if a voter can participate on a vote. It assumes the queried vote exists.
+    * @return True if the given voter can participate a certain vote, false otherwise
+    */
+    function _canVote(Vote storage vote_, address _voter) internal view returns (bool) {
+        return _isVoteOpen(vote_) && _canProceedAgreementAction(vote_.actionId) && token.balanceOfAt(_voter, vote_.snapshotBlock) > 0;
```

Comment:
On second thoughts this might not work actually. I've looked at Agreements more and afaik `_canProceedAgreementAction(vote_.actionId)` will return false if an Action is voided. But in the case of this app we still want to be able to vote if an Action is voided.